frontend/app/services/IdentityService.scala

```diff
@@ -44,80 +99,40 @@ case class IdentityService(identityApi: IdentityApi) {
   def doesUserPasswordExist(identityRequest: IdentityRequest): Future[Boolean] =
     identityApi.getUserPasswordExists(identityRequest.headers, identityRequest.trackingParameters)
 
-  def updateUserFieldsBasedOnJoining(user: IdMinimalUser, formData: CommonForm, identityRequest: IdentityRequest) {
-
-    val billingDetails = formData match {
-      case billingForm : PaidMemberJoinForm =>
-        val billingAddressForm = billingForm.billingAddress.getOrElse(billingForm.deliveryAddress)
-        billingAddress(billingAddressForm)
-      case _ => Json.obj()
-    }
-
-    val deliverAddress = formData match {
-      case jf : JoinForm => deliveryAddress(jf.deliveryAddress)
-      case _ => Json.obj()
-    }
-
-    val fields = Json.obj(
-      "secondName" -> formData.name.last,
-      "firstName" -> formData.name.first
-    ) ++ deliverAddress ++ billingDetails
-
-    val json = Json.obj("privateFields" -> fields)
-    postFields(json, user.id, identityRequest)
+  def updateUserPassword(password: String)(implicit idReq: IdentityRequest) {
+    identityApi.post("/user/password", Some(Json.obj("newPassword" -> password)), idReq.headers, idReq.trackingParameters, "update-user-password")(DisregardResponseContent)
   }
 
-  def updateUserPassword(password: String, identityRequest: IdentityRequest, userId: String) {
-    val json = Json.obj("newPassword" -> password)
-    identityApi.post("/user/password", Some(json), identityRequest.headers, identityRequest.trackingParameters, "update-user-password")
-  }
-
-  def updateUserMarketingPreferences(req: IdentityRequest, user: IdMinimalUser, allowMarketing: Boolean) =
-    postFields(Json.obj("statusFields.receiveGnmMarketing" -> allowMarketing), user.id, req)
```

Comment:
So `updateUserMarketingPreferences()` was actually unused anywhere in membership-frontend - it's been unused since https://github.com/guardian/membership-frontend/pull/1294 was merged last year. However, it's worth noting that I haven't yet added a _"Keep me up to date with offers from the Guardian"_ checkbox, and so I am _not_ setting `receiveGnmMarketing` to true when I create the user.

See the checkbox here on https://profile.theguardian.com/register :

![image](https://cloud.githubusercontent.com/assets/52038/25005219/e4b1e580-204f-11e7-8f11-aa1d40440135.png)
 

That checkbox will probably need to be added to the checkout on Membership Frontend before going fully live with this flow - it's still hid behind a feature flag tho', so I feel happy to leave that change to later.